lib/experimental/html/class-wp-html-tag-processor.php

```diff
@@ -1411,49 +1580,47 @@ public function __toString() {
 	 * @return string The processed HTML.
 	 */
 	public function get_updated_html() {
-		// Short-circuit if there are no updates to apply.
-		if ( ! count( $this->classname_updates ) && ! count( $this->attribute_updates ) ) {
+		if ( count( $this->classname_updates ) || count( $this->attribute_updates ) ) {
+			$this->apply_html_updates_and_stay_on_current_tag();
+
+			return $this->html;
+		} else {
 			return $this->updated_html . substr( $this->html, $this->updated_bytes );
 		}
+	}
 
-		/*
-		 * Parsing is in progress – let's apply the attribute updates without moving on to the next tag.
-		 *
-		 * In practice:
-		 * 1. Apply the attributes updates to the original HTML
-		 * 2. Replace the original HTML with the updated HTML
-		 * 3. Point this tag processor to the current tag name's end in that updated HTML
-		 */
+	/**
+	 * Applies all the enqueued attribute updates without advancing any
+	 * internal pointers to the next tag.
+	 *
+	 * @since 6.2.0
+	 */
+	private function apply_html_updates_and_stay_on_current_tag() {
```

Comment:
I have some questions about the need and use of this function.

can you share why we had to change it to what it is now?
I'm maybe having trouble seeing through the diff to what it's doing and why

Reply:
My largest goal was to avoid calling `get_updated_html()` in `seek()` as that name does not communicate how we're applying the updates and then lifting the parser to operate on the updated HTML.

As for the logic itself – we can keep it as it was, I just looked at it and thought it could do the same thing in less steps.

Reply:
> that name does not communicate how we're applying the updates and then lifting the parser to operate on the updated HTML.

this doesn't seem problematic to me. that we have to update the internal state when seeking isn't something that's exposed to the calling code, and it's relevant. it's an implementation detail and we could change it.

to me `seek()` only implies one thing: resetting the pointer to a different location, which is what it does. internally I'm okay with having this code acknowledge that _in the particular approach we've taken to accomplish that goal_ we have to flush all the current known updates.

Reply:
Oh I meant the „get_updated_html” name, not the „seek” name. It quacks like a getter, but it isn’t one.

Reply:
I'm still not following. how is `get_updated_html()` not like a getter?

Reply:
The name suggests it simply returns a value, but in reality it mutates the object. That's typically a domain of functions with names like `set_*`, `do_*`, `update_*`, `mutate_*` etc.

Reply:
I'm not really in agreement here; from the outside it doesn't matter because we don't expose that internal accounting. `get_updated_html()` is idempotent and safe to call. If it isn't then that's a bug, and we haven't exposed a way to extract those internals so as to work around the system.

Do you think it's possible this is a reactive response to a different issue?

Reply:
🤔 I guess you’re right, let’s just keep it as it was then.